pkg/operator/observe_config.go

```diff
@@ -292,6 +294,50 @@ func ConvertJSON(o interface{}) (interface{}, error) {
 	return ret, nil
 }
 
+// observeBuildAdditionalCA observes if the additional trusted CA is available.
+func observeBuildAdditionalCA(listers Listers, observedConfig map[string]interface{}, operatorConfig *operatorconfigv1alpha1.OpenShiftControllerManagerOperatorConfig) (map[string]interface{}, error) {
+	build, err := listers.buildConfigLister.Get("cluster")
+	if errors.IsNotFound(err) {
+		operatorConfig.Spec.AdditionalTrustedCA = nil
+		return observedConfig, nil
+	}
+	if err != nil {
+		return nil, err
+	}
+	if len(build.Spec.AdditionalTrustedCA.Name) == 0 {
+		operatorConfig.Spec.AdditionalTrustedCA = nil
+		return observedConfig, nil
+	}
+	cm, err := listers.clusterConfigConfigMapLister.ConfigMaps(openshiftConfigNamespaceName).Get(build.Spec.AdditionalTrustedCA.Name)
+	if errors.IsNotFound(err) {
+		operatorConfig.Spec.AdditionalTrustedCA = nil
+		glog.Warningf("referenced configMap %s/%s for additional trusted certificate authorities does not exist",
+			openshiftConfigNamespaceName,
+			build.Spec.AdditionalTrustedCA.Name)
+		return observedConfig, nil
+	}
+	if err != nil {
+		return nil, err
+	}
+	caData, err := mergePEMData(cm.Data)
+	if err != nil {
+		return nil, err
+	}
+	if len(caData) == 0 {
+		operatorConfig.Spec.AdditionalTrustedCA = nil
+		return observedConfig, nil
+	}
+	h := sha1.New()
+	h.Write(caData)
+	caHash := fmt.Sprintf("%x", h.Sum(nil))
+	operatorConfig.Spec.AdditionalTrustedCA = &operatorconfigv1alpha1.AdditionalTrustedCA{
+		SHA1Hash:      caHash,
+		ConfigMapName: cm.Name,
+	}
+	unstructured.SetNestedField(observedConfig, "/var/run/configmaps/additional-ca/additional-ca.crt", "build", "additionalTrustedCA")
```

Comment:
Additional CA's for the build controller need two pieces of config in `Spec`:

1. The hash and reference to the ConfigMap with the CA data
2. The mount location for the build controller's configuration.